### ISSUE TITLE
Question

### ISSUE THREAD

**Issue by PixelzChris** [korado531m7/InventoryMenuAPI#7]
Is transaction event works???
Like im getting a bug...
Every time im,changing windows my mc crashes... what I mean is (some times my hotbar disappears or its getting invisible)
HELP

**Comment by korado531m7** [comment]
transaction event won't be called
so you must deal with InventoryClickEvent or InventoryCloseEvent.
can you send me your plugin?

**Comment by PixelzChris** [comment]
Why do you want the plugin XD

Στις Τρί, 5 Μαρ 2019, 16:14 ο χρήστης korado531m7 <notifications@github.com>
έγραψε:

**Comment by PixelzChris** [comment]
Btw can I use both... your click event to change windows and the
transaction event for other things...?

Στις Τρί, 5 Μαρ 2019, 16:39 ο χρήστης Pixelz Chris <xristos.ale714@gmail.com>
έγραψε:

**Comment by korado531m7** [comment]
i wanna check the issue with your plugin

both of events can't change inventory data (e.g. inventory name).
you need to send per an invenory

**Comment by PixelzChris** [comment]
I sent the plugin it doesnt.have everything but you can see that there is a bug

**Comment by PixelzChris** [comment]
name: MenuTest
api: 3.0.0
main: Pixelz\teleporter
version: 0.5
author: Pixelz
commands:
  tps:
    description: Teleport System

Here is plugin.yml

Στις Τρί, 5 Μαρ 2019, 16:14 ο χρήστης korado531m7 <notifications@github.com>
έγραψε:

**Comment by korado531m7** [comment]
where did you upload the plugin?
i can't find that.

**Comment by PixelzChris** [comment]
On provite.email

Στις Παρ, 8 Μαρ 2019, 05:20 ο χρήστης korado531m7 <notifications@github.com>
έγραψε:

**Comment by PixelzChris** [comment]
Nvm what's your email,so I can send you the plugin...?

Στις Παρ, 8 Μαρ 2019, 07:33 ο χρήστης Pixelz Chris <xristos.ale714@gmail.com>
έγραψε:

**Comment by korado531m7** [comment]
support.system@icloud.com

**Comment by korado531m7** [comment]
please use InventoryClickEvent to check clicked item.
if you want to, i can make for it.

**Comment by PixelzChris** [comment]
Yea do it please I dont know how it works... btw I think there was an error

Στις Σάβ, 9 Μαρ 2019, 05:07 ο χρήστης korado531m7 <notifications@github.com>
έγραψε:

**Comment by korado531m7** [comment]
i sent you a plugin.

**Comment by PixelzChris** [comment]
Tu but when I change the click event it says... error class
korado531m7\InventoryMenuAPI.........\InventoryClickEvent does not exist
....

Στις Σάβ, 9 Μαρ 2019, 13:34 ο χρήστης korado531m7 <notifications@github.com>
έγραψε:

**Comment by PixelzChris** [comment]
Nvm I fixed it but When I click the item it appears on my inventory -_-
In transaction that is not happening

Στις Σάβ, 9 Μαρ 2019, 18:02 ο χρήστης Pixelz Chris <xristos.ale714@gmail.com>
έγραψε:

**Comment by korado531m7** [comment]
well, you must capture items with InventoryClickEvent.
you can't get the item from InventoryTransactionEvent.

**Comment by PixelzChris** [comment]
What if I dont want.the menu to,close... like I want to use a setting which
must be clickable and turn it in something else but it doesnt work btw the
window hardly appears...

Στις Κυρ, 10 Μαρ 2019, 06:41 ο χρήστης korado531m7 <notifications@github.com>
έγραψε:

**Comment by korado531m7** [comment]
use setReadonly function

**Comment by PixelzChris** [comment]
But is the click event going to work??
How do I do that

Στις Δευ, 11 Μαρ 2019, 08:38 ο χρήστης korado531m7 <notifications@github.com>
έγραψε:

**Comment by korado531m7** [comment]
no.
do i need to call clickevent in spite of setReadonly is false?

**Comment by PixelzChris** [comment]
Yes so you can click the item and it wont appear in your inventory... but click event will work

**Comment by korado531m7** [comment]
i'll add the function

**Comment by PixelzChris** [comment]
I,think we can add the code to remove the item on click event...

>

**Comment by korado531m7** [comment]
yeah, but i need to do that.

**Comment by PixelzChris** [comment]
How much time will it get..??
I have to finish my plugin asap....


Στις Δευ, 11 Μαρ 2019, 12:10 ο χρήστης korado531m7 <notifications@github.com>
έγραψε:

**Comment by korado531m7** [comment]
uploaded

**Comment by PixelzChris** [comment]
So whats new??

Στις Δευ, 11 Μαρ 2019, 18:02 ο χρήστης korado531m7 <notifications@github.com>
έγραψε:

**Comment by korado531m7** [comment]
https://github.com/korado531m7/InventoryMenuAPI/commit/b7a18b81f30db72c1d6518154b3b906e5b9068ab

**Comment by PixelzChris** [comment]
Error: Call to a member function getContents() on null
File: plugins/Teleporter/src/korado531m7/InventoryMenuAPI/InventoryMenu
Line: 170
Type: Error

THIS CRASH WAS CAUSED BY A PLUGIN
BAD PLUGIN: TeleportSystem v1.0.0

Code:
[161]         $tag = new CompoundTag();
[162]         $tag->setString('CustomName', $this->getName());
[163]         IMU::sendTagData($player, $tag, $pos);
[164]     }
[165]
[166]     /**
[167]      * this function is for internal use only. Don't call this
[168]      */
[169]     public function setData(Player $player, $inv){
[170]         InventoryMenuAPI::setData($player, $this, $this->getName(),
$inv, $player->getInventory()->getContents());
[171]     }
[172] }
When I click an item

Στις Δευ, 11 Μαρ 2019, 19:52 ο χρήστης korado531m7 <notifications@github.com>
έγραψε:

**Comment by PixelzChris** [comment]
Btw some times the item gets on my hot bar but it does disappear until i click it

**Comment by PixelzChris** [comment]
Btw id code on click event doesnt work
}elseif ($item->getId....

**Comment by korado531m7** [comment]
please show me your code

**Comment by PixelzChris** [comment]
```public function onClicked(InventoryClickEvent $event) {
        $player = $event->getPlayer();
        $item = $event->getItem();
        if ($item->getName() == C::GREEN . "GUI") {
            if (count($this->getServer()->getOnlinePlayers()) < 52) {
                $player->sendMessage(C::GREEN . "Selected GUI. Now select the player you wish to tp to.");
                GUI::openTpgui($player);
            } else {
                $this->closeInventory($player);
                $player->sendMessage(C::RED . "More than 52 players are not supported yet...");
                }
        }elseif ($item->getId() == 397 && $player->hasPermission("tpgui.cmd")){
			        if(($p = Server::getInstance()->getPlayerExact(C::clean($item->getName()))) instanceof Player){
                    $player->sendMessage(C::GRAY . "You have been" . C::GREEN . " successfully " . C::GRAY . "teleported!");
                    $player->teleport($p);
                }
        }
        if ($item->getName() == C::RED . C::BOLD . "Close the menu\n\n" . C::RESET . C::YELLOW . "Click here!"){
             }```

**Comment by PixelzChris** [comment]
The id part is not working

if($item->getId() == 397 ){

It doesnt work...
Like im clicking a 397 id item and it does nothing

**Comment by korado531m7** [comment]
why don't you use === instead of == ?
or, use var_dump

**Comment by PixelzChris** [comment]
Error: Call to a member function getContents() on null
File: plugins/TeleportSystem_v1.0.0/src/korado531m7/InventoryMenuAPI/InventoryMenu
Line: 169
Type: Error

THIS CRASH WAS CAUSED BY A PLUGIN
BAD PLUGIN: TeleportSystem v1.0.0

Code:
[160]         $tag = new CompoundTag();
[161]         $tag->setString('CustomName', $this->getName());
[162]         IMU::sendTagData($player, $tag, $pos);
[163]     }
[164]     
[165]     /**
[166]      * this function is for internal use only. Don't call this
[167]      */
[168]     public function setData(Player $player, FakeMenuInventory $inv){
[169]         InventoryMenuAPI::setData($player, $this, $inv, $player->getInventory()->getContents());
[170]     }
[171] }
